huaweicloud/services/iam/data_source_huaweicloud_identity_group.go

```diff
@@ -0,0 +1,181 @@
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"password_expires_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"password_status": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"password_strength": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func DataSourceIdentityGroupV3Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	identityClient, err := config.IdentityV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud identity client: %s", err)
+	}
+
+	listOpts := groups.ListOpts{
+		Name: d.Get("name").(string),
+	}
+
+	allPages, err := groups.List(identityClient, listOpts).AllPages()
+	if err != nil {
+		return fmtp.Errorf("Unable to query groups: %s", err)
+	}
+
+	allGroups, err := groups.ExtractGroups(allPages)
+
+	if err != nil {
+		return fmtp.Errorf("Unable to retrieve groups: %s", err)
+	}
+
+	conditions := map[string]interface{}{}
+
+	if v, ok := d.GetOk("id"); ok {
+		conditions["id"] = v.(string)
+	}
+	if v, ok := d.GetOk("description"); ok {
+		conditions["description"] = v.(string)
+	}
+
+	var foundGroups []groups.Group
+
+	for _, group := range allGroups {
+		if groupsFilter(group, conditions) {
+			foundGroups = append(foundGroups, group)
+		}
+	}
+
+	if len(foundGroups) < 1 {
+		return fmtp.Errorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+
+	if len(foundGroups) > 1 {
+		return fmtp.Errorf("Your query returned more than one result. " +
+			"Please try a more specific search criteria.")
+	}
+
+	group := foundGroups[0]
+
+	d.SetId(group.ID)
+
+	mErr := multierror.Append(nil,
+		d.Set("domain_id", group.DomainID),
+		d.Set("name", group.Name),
+	)
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.Errorf("error setting identity group fields: %s", err)
+	}
+
+	d.Set("description", group.Description)
+
+	// get users of this group
+	allUsers, err := groups.ListUsers(identityClient, d.Id()).Extract()
+
+	if err != nil {
+		return fmtp.Errorf("Unable to retrieve users: %s", err)
+	}
+
+	if len(allUsers) > 0 {
+		users := make([]map[string]interface{}, 0, len(allUsers))
+		for _, user := range allUsers {
+			userToSet := setUserAttributes(user)
+			users = append(users, userToSet)
+		}
+
+		d.Set("users", users)
+	}
+
+	return nil
+}
+
+func groupsFilter(group groups.Group, conditions map[string]interface{}) bool {
+	if v, ok := conditions["id"]; ok && v != group.ID {
+		return false
+	}
+	if v, ok := conditions["description"]; ok && v != group.Description {
+		return false
+	}
+	return true
+}
+
+func setUserAttributes(user groups.User) map[string]interface{} {
+	userToSet := make(map[string]interface{})
```

Comment:
the type of this variable is Map, but the name sounds like a Set, please make it easy to understanding.

Reply:
fixed